lib/account/account-service.js

```diff
@@ -104,16 +133,23 @@ class AccountService {
      *
      * TODO: Encrypt data
      *
-     * @param {string} accountId
      * @param {Object} accountInfo
      * @param {Object} options
-     * @return {Promise.<AccountService~initiateResult>}
+     * @return {Promise.<accountManager~initiationInfo>}
      */
-    initiateAsync(directory, accountInfo, options) {
-        return this.storage.putAsync(directory, JSON.stringify(accountInfo), options).then(() => {
-            return accountInfo;
+    function signupInitiateAsync(accountInfo, options) {
+        return getDirectoryAsync(registrationDirName, accountInfo.regId).then((directory) => {
+            return storage.putAsync(directory, JSON.stringify(accountInfo), options).then(() => {
```

Comment:
I would prefer you put the next `.then()` outside.

```
return getDirectoryAsync(registrationDirName, accountInfo.regId).then((directory) => {
    return storage.putAsync(directory, JSON.stringify(accountInfo), options);
}).then(() => {
    return {
        regId.accountInfo.regId;
    };
});
```

This will help prevent our code from becoming terribly indented.